src/splat/segtypes/segment.py

```diff
@@ -447,9 +447,8 @@ def symbol_name_format_no_rom(self) -> str:
     @property
     def subalign(self) -> Optional[int]:
         if self.parent:
-            return self.parent.subalign
-        else:
-            return self.given_subalign
+            log.error("subalign is not valid for non-top-level segments")
```

Comment:
This feels more like a development error than a user error, so I think we should use an assert instead.
An assert would give us a traceback which should be easier to debug
```suggestion
            assert self.parent is None, f"subalign is not valid for non-top-level segments. ({self})"
```